Source/CarShack/Hypermedia/Customers/FavoriteCustomer.cs

```diff
@@ -1,12 +1,14 @@
 ﻿using System.ComponentModel.DataAnnotations;
 using WebApi.HypermediaExtensions.Hypermedia.Actions;
+using WebApi.HypermediaExtensions.JsonSchema;
+using WebApi.HypermediaExtensions.WebApi.RouteResolver;
 
 namespace CarShack.Hypermedia.Customers
 {
     public class FavoriteCustomer : IHypermediaActionParameter
     {
         [Required]
-        [Url]
-        public string CustomerLink { get; set; }
+        [KeyFromUri(typeof(HypermediaCustomer), schemaProperyName: "Customer", routeTemplateParameterName: null)]
```

Comment:
Is "routeTemplateParameterName: null" required to say: Just take the only one there is?

Reply:
Yes, and there is an overload without this parameter. I updated the sample.